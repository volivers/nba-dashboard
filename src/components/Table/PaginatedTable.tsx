import React from 'react';
import { Pagination, Table } from 'react-bootstrap';

export interface PaginatedTableProps {
  children: React.ReactNode;
  description: string;
  isLastPage: boolean;
  onPageChange: (page: number) => void;
  page: number;
  title: string;
}

const PaginatedTable = ({
  children,
  description,
  isLastPage,
  onPageChange,
  page,
  title,
}: PaginatedTableProps) => (
  <div>
    <h1 className="my-3">{title}</h1>
    <div className="d-flex justify-content-between">
      <p><small>{description}</small></p>
      <Pagination className="mt-3">
        <Pagination.Prev
          disabled={page === 1}
          onClick={() => onPageChange(page - 1)}
        />
        <Pagination.Next
          disabled={isLastPage}
          onClick={() => onPageChange(page + 1)}
        />
      </Pagination>
    </div>
    <Table hover responsive>
      {children}
    </Table>
  </div>
);

export default PaginatedTable;
