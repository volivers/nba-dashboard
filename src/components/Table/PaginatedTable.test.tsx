import { render, screen } from '@testing-library/react';
import PaginatedTable, { PaginatedTableProps } from './PaginatedTable';

describe('PaginatedTable', () => {
  let props: PaginatedTableProps;

  beforeEach(() => {
    props = {
      children: (<div>Component</div>),
      description: 'Lorem ipsum',
      isLastPage: false,
      onPageChange: jest.fn(),
      page: 1,
      title: 'Title',
    };
  });

  it('renders properly', () => {
    const { container } = render(<PaginatedTable {...props} />);

    expect(container).toMatchSnapshot();
  });

  describe('when the first page is provided', () => {
    beforeEach(() => {
      props = {
        ...props,
        page: 1,
      };
    });

    it('disables the previous button', () => {
      render(<PaginatedTable {...props} />);

      expect(screen.getAllByRole('listitem')[0]).toHaveClass('disabled');
    });
  });

  describe('when the last page is provided', () => {
    beforeEach(() => {
      props = {
        ...props,
        isLastPage: true,
      };
    });

    it('disables the next button', () => {
      render(<PaginatedTable {...props} />);

      expect(screen.getAllByRole('listitem')[1]).toHaveClass('disabled');
    });
  });
});
