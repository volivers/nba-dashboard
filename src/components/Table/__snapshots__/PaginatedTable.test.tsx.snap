// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatedTable renders properly 1`] = `
<div>
  <div>
    <h1
      class="my-3"
    >
      Title
    </h1>
    <div
      class="d-flex justify-content-between"
    >
      <p>
        <small>
          Lorem ipsum
        </small>
      </p>
      <ul
        class="mt-3 pagination"
      >
        <li
          class="page-item disabled"
        >
          <span
            class="page-link"
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover"
      >
        <div>
          Component
        </div>
      </table>
    </div>
  </div>
</div>
`;
